tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/field/field_datatype_test.dart

```diff
@@ -74,19 +83,174 @@ void main() {
+          table: serverpod_table
+          fields:
+            nickname: String
+          ''',
+        ).build(),
+        ModelSourceBuilder().withYaml(
+          '''
+          class: Example
+          fields:
+            name: module:serverpod:ServerpodClass
+          ''',
+        ).build()
+      ];
+
+      var collector = CodeGenerationCollector();
+      StatefulAnalyzer analyzer = StatefulAnalyzer(
+        models,
+        onErrorsCollector(collector),
+      );
+      var definitions = analyzer.validateAll();
+
+      test('then no errors was generated', () {
+        expect(collector.errors, isEmpty);
+      });
+
+      test('then the class name is the one defined in serverpod.', () {
+        var definition = definitions.first as ClassDefinition;
+        expect(definition.fields.first.type.className, 'ServerpodClass');
+      });
+    });
+
+    group(
+        'Given a class with a field with a serverpod class type referenced by module prefix',
+        () {
+      var models = [
+        ModelSourceBuilder().withModuleAlias('serverpod').withYaml(
+          '''
+          class: ServerpodClass
+          table: serverpod_table
+          fields:
+            nickname: String
+          ''',
+        ).build(),
+        ModelSourceBuilder().withYaml(
+          '''
+          class: Example
+          fields:
+            name: serverpod:ServerpodClass
+          ''',
+        ).build()
+      ];
+
+      var collector = CodeGenerationCollector();
+      StatefulAnalyzer analyzer = StatefulAnalyzer(
+        models,
+        onErrorsCollector(collector),
+      );
+      var definitions = analyzer.validateAll();
+
+      test('then no errors was generated', () {
+        expect(collector.errors, isEmpty);
+      });
+
+      test('then the class name is the one defined in serverpod.', () {
+        var definition = definitions.first as ClassDefinition;
+        expect(definition.fields.first.type.className, 'ServerpodClass');
+      });
+    });
+
+    test(
+        'Given a class with a field with a module type without the module:alias path then an error is reported that the datatype does not exist.',
+        () {
+      var models = [
+        ModelSourceBuilder().withModuleAlias('auth').withYaml(
+          '''
+          class: UserInfo
+          table: serverpod_user_info
+          fields:
+            nickname: String
+          ''',
+        ).build(),
+        ModelSourceBuilder().withYaml(
+          '''
+          class: Example
+          fields:
+            name: UserInfo
+          ''',
+        ).build()
+      ];
+
+      var collector = CodeGenerationCollector();
+      StatefulAnalyzer analyzer = StatefulAnalyzer(
+        models,
+        onErrorsCollector(collector),
+      );
+      analyzer.validateAll();
+
+      expect(collector.errors, isNotEmpty);
+      expect(
+        collector.errors.first.message,
+        contains('The field has an invalid datatype "UserInfo".'),
```

Comment:
Question: Do we have information to give a hint that there might exists a data type with a module prefix? Could be helpful especially in the LSP.

Reply:
(Is probably a separate feature/PR)

Reply:
Question 2: For my knowledge - We have always only supported referencing module with the module prefix right?

Reply:
1: It would be possible but then we have to search in all other models too, yeah another PR/feature I would say.
2: Correct.